core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -98,14 +109,13 @@ protected String name() {
     Preconditions.checkArgument(namespace.levels().length >= 1,
         "Missing database in table identifier: %s", namespace);
 
-    Joiner slash = Joiner.on("/");
-    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
```

Comment:
The outer `SLASH.join` is unnecessary because a base path can be passed in: `new Path(base, relative)`.